### PR TITLE
FIXES: alsamixer lowering volume control widget past 0 causes hostile volume spike to 100

### PR DESCRIPTION
**Pull request by genbtc**
bug: lowering volume below 0 causes volume to overflow and spike to 100% volume hurting my ears.

to reproduce:  when holding down z,x,c to lower the volume, the volume spikes to 100% and then down to 0% over and over

background:
when volume is already 0 and direction is -1 (lowering volume) - but lower than 0 is impossible
the function `set_normalized_volume` @ volume-mapping.c is eventually still called anyway
`value = lrint_dir(6000.0 * log10(volume), dir) + max;`
and volume is 0, so
log10(0) resolves to negative infinity, so
value overflows past infinity. 9223372036854775802
alsamixer-alsa-utils-volumemap-bug3_2024-04-18_13-27-23

my fix:
this function `change_volume_relative` function will now skip execution of the set_normalized_playback_volume function when volume is 0 and the dir is -1 
TLDR: (stop trying to set a lower volume than 0)

### CODE REVIEW

**Comment by perexg** [comment]
I think that this change is better (fixes the function where the problem occurs and tries to set the minimal dB value again instead skipping the command):

```
diff --git a/alsamixer/volume_mapping.c b/alsamixer/volume_mapping.c
index 29ab061..f34801c 100644
--- a/alsamixer/volume_mapping.c
+++ b/alsamixer/volume_mapping.c
@@ -146,6 +146,8 @@ static int set_normalized_volume(snd_mixer_elem_t *elem,
                min_norm = pow(10, (min - max) / 6000.0);
                volume = volume * (1 - min_norm) + min_norm;
        }
+       if (volume <= 0)
+               volume = 1e-36;
        value = lrint_dir(6000.0 * log10(volume), dir) + max;
        return set_dB[ctl_dir](elem, channel, value, dir);
 }
```
